wagtail/admin/views/generic/models.py

```diff
@@ -256,13 +256,29 @@ def get_buttons(column, instance, *args, **kwargs):
         )
 
     def _get_custom_column(self, field_name, column_class=Column, **kwargs):
-        label, attr = label_for_field(field_name, self.model, return_attr=True)
+        *relation, field = field_name.split(".")
+        model_class = self.model
+
+        # iterate over the relation list to try to get the last model
+        # where the field exists
+        for model in relation:
+            model_class = getattr(model_class, "_meta").get_field(model).related_model
+            model_name = model
+
+        label, attr = label_for_field(field, model_class, return_attr=True)
+
+        # add the name of relational model to make the label more clear
+        try:
+            label = model_name + " " + label
+        except NameError:
+            pass
```

Comment:
I don't think this is necessary. Or if we want to, we definitely don't want to use the field's internal name directly here, which is a code string in snake_case and should not be exposed to the user. A better idea would be to use the model's verbose name (`model._meta.verbose_name`)

Reply:
Changed to model's verbose name, can be removed if not necessary.